src/checkpoint.hh

```diff
@@ -0,0 +1,1246 @@
+/* {{{
+
+    This file is part of gapc (GAPC - Grammars, Algebras, Products - Compiler;
+      a system to compile algebraic dynamic programming programs)
+
+    Copyright (C) 2008-2011  Georg Sauthoff
+         email: gsauthof@techfak.uni-bielefeld.de or gsauthof@sdf.lonestar.org
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+    * Author: fymue
+}}} */
+
+#ifndef SRC_CHECKPOINT_HH_
+#define SRC_CHECKPOINT_HH_
+
+#include <string>
+#include <array>
+#include <vector>
+#include <list>
+#include <utility>
+#include <sstream>
+#include "printer.hh"
+#include "symbol.hh"
+#include "statement/table_decl.hh"
+#include "type/base.hh"
+
+typedef hashtable<std::string, Symbol::NT*> nt_tables;
+
+namespace Printer {
+class Checkpoint : public Base {
+ private:
+  // true if table types are wrapped in ListRef
```

Comment:
please comment why this makes a difference

Reply:
Extending from `Base` gives me access to the `indent` and `inc_indent` / `dec_indent` methods and ensures that the indent matches the indent in the `Printer::Cpp::print...` methods.